### PR TITLE
Rename resolver/cache.Operator to Entry.

### PR DESCRIPTION
**Pull request by benluddy**
This type is a data object representing a resolver cache entry. It
doesn't have a particular connection to operators other than residing
where it resides. This name change should clarify what, exactly, the
type represents.

### CODE REVIEW

File: pkg/controller/registry/resolver/cache/predicates.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ import (
 )
 
 type OperatorPredicate interface {
```
Comment (njhale) [inline pkg/controller/registry/resolver/cache/predicates.go:13]: Might as well rename this `Predicate` while we're at it.

**Review by njhale — APPROVED** [review]
/lgtm

**Comment by timflannagan** [comment]
Known issue with the e2e tests that seems to be propping up more recently. Feel free to just manually squash merge this but holding so the bot doesn't go crazy.

/hold

**Comment by benluddy** [comment]
@timflannagan Does this need to be rebased in order to get the E2E check passing?

**Comment by benluddy** [comment]
https://github.com/operator-framework/operator-lifecycle-manager/issues/2092

**Pull request merged**